File: server/features/window.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// WindowFocus describes the FocusWindow RPC that brings the app window to the foreground.
var WindowFocus = featureregistry.Feature{
	ID:          "window-focus",
	Title:       "Focus Window",
	Description: "Brings the application window to the foreground on the host platform.",
	RPCIDs:      []string{"window:focus"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(WindowFocus)
}

File: web-app/src/lib/features/features/window.ts
import type { Feature } from '../types';

export const windowFeatures = {
  'window-focus': {
    id: 'window-focus',
    title: 'Focus Window',
    description: 'Brings the application window to the foreground on the host platform.',
    rpcIds: ['window:focus'],
    componentPaths: [],
    testIds: [
      'TestFocusWindow_EmptyID',
      'TestFocusWindow_NonDarwinPlatform',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
